bi-connectors/TableauConnector/opensearch_sql_jdbc/dialect.tdd

```diff
@@ -58,10 +58,15 @@
         </function>
 
         <function group='logical' name='IFNULL' return-type='datetime'>
-            <formula>CASE WHEN ISNULL(%1) THEN SUBDATE(TIMESTAMP(%2), INTERVAL HOUR(TIMESTAMP(%2)) HOUR) ELSE %1 END</formula>
+            <formula>CASE WHEN ISNULL(%1) THEN CAST(DATE_FORMAT(%2, '%Y-%m-%d 00:00:00') AS TIMESTAMP) ELSE %1 END</formula>
```

Comment:
Is it the bug of TIMESTAMP()?

Reply:
This is not a a bug of TIMESTAMP(). The original query did not work because of a type mismatch. I did cast to TIMESTAMP to have the same data types.
Original query with error:
```
curl -XPOST http://localhost:9200/_plugins/_sql -u admin:admin -H 'Content-Type: application/json' -d '{"query": "SELECT CASE WHEN ISNULL(`calcs`.`date0`) THEN SUBDATE(TIMESTAMP(DATE(\"2010-04-12\")), INTERVAL HOUR(TIMESTAMP(DATE(\"2010-04-12\"))) HOUR) ELSE `calcs`.`date0` END FROM `calcs`"}'
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "All result types of CASE clause must be the same, but found [DATETIME, TIMESTAMP]",
    "type": "SemanticCheckException"
  },
  "status": 400 
```